pkg/cmd/get.go

```diff
@@ -0,0 +1,182 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+)
+
+type GetOptions struct {
+	AllNamespaces  bool
+	IsCoreResource bool
+	APIPath        string
+	Resource       string
+	Token          string
+
+	KubeArchiveHost string
+
+	RESTConfig *rest.Config
+	kubeFlags  *genericclioptions.ConfigFlags
+}
+
+func NewGetOptions() *GetOptions {
+	return &GetOptions{
+		kubeFlags:       genericclioptions.NewConfigFlags(true),
+		KubeArchiveHost: "https://localhost:8081",
+	}
+}
+
+func NewGetCmd() *cobra.Command {
+	o := NewGetOptions()
+
+	cmd := &cobra.Command{
+		Use:   "get [RESOURCES]",
+		Short: "Command to get resources from KubeArchive",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			err = o.Validate(args)
```

Comment:
Instead of doing the validation of the args for this function in the `RunE` method, you could use [`cobra.Command.Args`](https://pkg.go.dev/github.com/spf13/cobra#Command.Args) ([see this example](https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#positional-and-custom-arguments)). All you would need to do is add a `*cobra.Cmd` arg to `GetOptions.Validate` so that it matches the type for [`cobra.PositionalArgs` ](https://pkg.go.dev/github.com/spf13/cobra#PositionalArgs)

Reply:
I changed it by `cobra.ExactArgs(1)`, for now that will suffice. The moment we need to validate that the args have a certain pattern to them, I think we will need to extract it back.